app/diagnostics.go

```diff
@@ -79,14 +82,37 @@ func (a *App) sendDailyDiagnostics(override bool) {
 		a.trackGroups()
 		a.trackChannelModeration()
 	}
+
+	if *a.Config().LogSettings.EnableDiagnostics && a.IsLeader() && ((!strings.Contains(RUDDER_KEY, "placeholder") && !strings.Contains(RUDDER_DATAPLANE_URL, "placeholder")) || override) {
```

Comment:
This has a lot of duplication with the previous condition. I would suggest refactoring it to something like:

```go
if *a.Config().LogSettings.EnableDiagnostics && a.IsLeader() {
	enableSegment := !strings.Contains(SEGMENT_KEY, "placeholder")
	enableRudder := !strings.Contains(RUDDER_KEY, "placeholder") && !strings.Contains(RUDDER_DATAPLANE_URL, "placeholder")

	if enableSegment || override {
		a.Srv().initDiagnostics("")
	}
	if enableRudder || override {
		a.Srv().initRudder(RUDDER_DATAPLANE_URL)
	}

	if enableSegment || enableRudder || override {
	   // common stuff here
	}
} 
```

Reply:
Ditto - duplication is intentional as the first one will be deleted entirely in the next version,.

Reply:
Sounds good. Probably good to add a comment explaining the reason for the duplication.

Reply:
Will leave it as the PR to remove this will be landing soon afterwards!